distribution_script/router.js

```diff
@@ -0,0 +1,103 @@
+const express = require('express');
+const fs = require('fs');
+const path = require('path');
+const moment = require('moment');
+const { BN, fromWei } = require('web3').utils;
+const { poolNames, pools, PRIVATE_OFFERING, REWARD_FOR_STAKING } = require('./constants');
+const contracts = require('./contracts');
+
+const router = express.Router();
+
+function checkNumberOfInstallments(db, pool) {
+    let error = null;
+    let secondsFromCliff = Date.now() / 1000 - (db.distributionStartTimestamp + db.cliff[pool] * db.stakingEpochDuration);
+    if (secondsFromCliff < 0) {
+        secondsFromCliff = 0;
+    }
+    let expectedNumberInstallmentsMade = Math.floor(secondsFromCliff / db.stakingEpochDuration);
+    expectedNumberInstallmentsMade = Math.min(expectedNumberInstallmentsMade, db.numberOfInstallments[pool]);
+    if (expectedNumberInstallmentsMade > db.numberOfInstallmentsMade[pool] + 1) {
```

Comment:
What is the purpose of `+ 1` here? Why can't we just compare `expectedNumberInstallmentsMade` with `db.numberOfInstallmentsMade[pool]`?

Reply:
Because it will conflict with https://github.com/poanetwork/dpos-token/pull/33/files/5948f58984f447c4399fb7fe6d9c731d55c253ea#diff-737b671c8378042f06b68abd82381c24R80

Reply:
I can also add validation for this case:
```
if (db.numberOfInstallmentsMade[pool] > expectedNumberInstallmentsMade) {
    error = 'The current number of installments made is greater than expected'
}
```

Reply:
> Because it will conflict with

Ok, got it

> I can also add validation for this case

Yes, let's add that one as well.

Reply:
> Because it will conflict with...

Should we do the same for the `checkDistributedValue` function?

Reply:
Yes, we should

Reply:
I mean the `checkDistributedValue` also conflicts with https://github.com/poanetwork/dpos-token/blob/239ae082936863707644823e13ce221ca3235c49/distribution_script/router.js#L83, doesn't it?

Reply:
Sorry, I misunderstood you
No, it doesn't. In `checkDistributedValue` it calculates the `installment value` manually, multiplies it by the `number of installments made` from the contract and compares the result with the `distributed value` from contract (`stake - tokensLeft`). Just to make sure that distributed value is correct